lib/cloud_controller/diego/app_recipe_builder.rb

```diff
@@ -176,6 +176,12 @@ def generate_ssh_action(user, environment_variables)
       def generate_environment_variables(lrp_builder)
         environment_variables = lrp_builder.port_environment_variables
 
+        if process.app.lifecycle_type == BuildpackLifecycleDataModel::LIFECYCLE_TYPE
```

Comment:
We have been trying to not introduce lifecycle specific code into `AppRecipeBuilder`. If you wanted to add a method `#foo` to the`Buildpack:: DesiredLrpBuilder` one, we ask that there be a matching one on the `Docker::DesiredLrpBuilder` to avoid having branch statements on lifecycle type.